crypto/cipher_extra/cipher_test.cc

```diff
@@ -1454,3 +1454,188 @@ TEST(CipherTest, Empty_EVP_CIPHER_CTX_V1187459157) {
+      std::vector<uint8_t> plaintext(pt_len);
+      std::vector<uint8_t> aad;
+
+      RAND_bytes(key.data(), key_len);
+      if (iv_len > 0) {
+        RAND_bytes(iv.data(), iv_len);
+      }
+      if (pt_len > 0) {
+        RAND_bytes(plaintext.data(), pt_len);
+      }
+      if (is_aead) {
+        aad.resize(16);
+        RAND_bytes(aad.data(), aad.size());
+      }
+
+      // Encrypt
+      bssl::UniquePtr<EVP_CIPHER_CTX> ctx(EVP_CIPHER_CTX_new());
+      ASSERT_TRUE(ctx);
+      ASSERT_TRUE(EVP_EncryptInit_ex(ctx.get(), cipher, nullptr, nullptr, nullptr));
+      if (!is_xts) {
+        ASSERT_TRUE(EVP_CIPHER_CTX_set_padding(ctx.get(), 0));
+      }
+
+      if (is_ccm) {
+        ASSERT_TRUE(EVP_CIPHER_CTX_ctrl(ctx.get(), EVP_CTRL_AEAD_SET_TAG, 16, nullptr));
+      }
+
+      ASSERT_TRUE(EVP_EncryptInit_ex(ctx.get(), nullptr, nullptr, key.data(),
+                                     iv_len > 0 ? iv.data() : nullptr));
+
+      if (is_ccm) {
+        int len;
+        ASSERT_TRUE(EVP_CipherUpdate(ctx.get(), nullptr, &len, nullptr, pt_len));
+      }
+
+      if (!aad.empty()) {
+        int len;
+        ASSERT_TRUE(EVP_EncryptUpdate(ctx.get(), nullptr, &len, aad.data(), aad.size()));
+      }
+
+      size_t max_ct_len = pt_len + block_size;
+      std::vector<uint8_t> ciphertext(max_ct_len);
+      int out_len = 0;
+      if (pt_len > 0) {
+        ASSERT_TRUE(EVP_EncryptUpdate(ctx.get(), ciphertext.data(), &out_len,
+                                      plaintext.data(), pt_len));
+      }
+      size_t ct_len = out_len;
+
+      int final_len = 0;
+      ASSERT_TRUE(EVP_EncryptFinal_ex(ctx.get(), ciphertext.data() + ct_len, &final_len));
+      ct_len += final_len;
+      ciphertext.resize(ct_len);
+
+      std::vector<uint8_t> tag;
+      if (is_aead) {
+        tag.resize(16);
+        ASSERT_TRUE(EVP_CIPHER_CTX_ctrl(ctx.get(), EVP_CTRL_AEAD_GET_TAG,
+                                        tag.size(), tag.data()));
+      }
+
+      // Decrypt
+      ctx.reset(EVP_CIPHER_CTX_new());
+      ASSERT_TRUE(ctx);
+      ASSERT_TRUE(EVP_DecryptInit_ex(ctx.get(), cipher, nullptr, nullptr, nullptr));
+      if (!is_xts) {
+        ASSERT_TRUE(EVP_CIPHER_CTX_set_padding(ctx.get(), 0));
+      }
+
+      if (is_aead) {
+        ASSERT_TRUE(EVP_CIPHER_CTX_ctrl(ctx.get(), EVP_CTRL_AEAD_SET_TAG,
+                                        tag.size(), tag.data()));
+      }
+
+      ASSERT_TRUE(EVP_DecryptInit_ex(ctx.get(), nullptr, nullptr, key.data(),
+                                     iv_len > 0 ? iv.data() : nullptr));
+
+      if (is_ccm) {
+        int len;
+        ASSERT_TRUE(EVP_CipherUpdate(ctx.get(), nullptr, &len, nullptr, ct_len));
+      }
+
+      if (!aad.empty()) {
+        int len;
+        ASSERT_TRUE(EVP_DecryptUpdate(ctx.get(), nullptr, &len, aad.data(), aad.size()));
+      }
+
+      std::vector<uint8_t> decrypted(ct_len + block_size);
+      out_len = 0;
+      if (ct_len > 0) {
```

Comment:
when would `ct_len` be 0? in that case, would there still be an auth tag? if not, we can probably either skip this case earlier or at least avoid the allocation on L1623.